### ISSUE TITLE
App causes FritzBox to be very slow and wifi connections of the FritzBox to drop

### ISSUE THREAD

**Issue by Boechie** [puschie286/fritzbox-bridge#40]
For quite some time already my FritzBox 7560 is very slow. Opening the login page of the web interface takes about 30 seconds(!) and sometimes even times out. Today things got even worse. Wifi connection for all devices became very unstable, even dropping out every few minutes. As the wifi troubles caused urgency (the slow web interface was uncomfortable, but not urgent), I tracked down the cause of this problem. To my surprise disabling the FritzBox app in Homey solved both problems. Turning it back on reintroduced the problems. Hence made the causal connection between the problems and the FritzBox Homey app.

To me it seems that problems became bigger after the last update although the changelog tells me the contrary.

It helps if I increase the polling interval in the config of the app to higher than 20 secs. My webinterface of FritzBox now loads pages in about 8 secs, which is still very slow, but workable. With the app disabled it pages open in 5 secs. The wifi connection now stays stable with the 20 secs setting. If I go below 20 secs (ex. 10 secs) then my webinterface takes about 20 secs to load a page. Wifi is then so-so. If I go lower to 1 sec, then pages of the FritzBox webinterface take more than 30 secs or time out and wifi becomes really unstable and drops out.

I think somehow things don;t work optimally in the latest version. Even a polling interval of 1 sec shouldn't even create that high a load so that wifi connections drop. Could you look into this?

**Comment by puschie286** [comment]
Hi,
sorry that my app has caused or worsened these problems.
can you confirm which version of the FritzboxBridge you are currently using ?

Hmm, it sounds like your fritzbox is under heavy load and cant process the requests. 
(Other users reporting similar behavior found out it was due some heat problems or devices that causes high cpu load on the fritzbox)

in previous versions, the app waited until it receives an answer but that resulted sometimes in endless wait (until app/homey had restart). the current version limit this wait time to 10 seconds (which is not configurable at this point).
a few versions ago, i made the optional network device check non-optional because the cards doesnt work without it. (which might be the reason you see the load increase for newer version)

So i see two things that might help you:
1. option to disable the network device check or set a different polling interval
2. make the 10 seconds wait limit configurable

Note: the app just do 2 web request to the fritzbox and process the content (like when you open the webinterface by yourself) so there is not much i can do to speed this up

**Comment by puschie286** [comment]
Version v0.10.1 increased the response wait time to 40 seconds - this should lower the problem significant.

For more optimizations i would like to know if you use network related cards ? (Like "Network device has connected" or "Network device is connected/disconnected" )

**Comment by Boechie** [comment]
> For more optimizations i would like to know if you use network related cards ? (Like "Network device has connected" or "Network device is connected/disconnected" )

Yes, I use those cards a lot, because I built smart presence logic on those events and conditions. In our office lights, heating, etc. work based on this smart presence logic.

I changed the Data Query Interval a few months ago (probably back in March when I reported the problem) to 1 minute because of all the performance problems I experienced. Since then performance is OK. I would really like to use a shorter interval because now it might take 1 minute before the lights turn on when someone arrives near the office building... Also I'm missing quite some events (phones entering or leaving the wifi network). It seems that the trigger card is not always fired on (dis)connect events, **but I'm not sure that that is caused by the Fritzbox-app**! It could also be caused by other apps that I also used to program the smart presence flows/scripts.

Now a few minutes ago I changed the Data Query Interval to 5 secs (my desired setting). The Fritzbox UI gets slower with load times of 8-10 secs. Not ideal, but workable. The total CPU usage of Fritzbox doubles, but still below 50%, which seems OK to me.

Thank you for taking the time to look at mu issue! Please let me know if you have further improvements. I'm interested in testing them.

**Comment by puschie286** [comment]
The newest version 0.11.0 reduce the required requests to a minimum. This should improve the network update performance for older/slower fritzbox a lot.

I would also suggest to remove the fritzbox device in homey if you have it added - this removes the os version change check.

This should maximize your performance - further improvements might not be possible with http (api) requests. The performance of these requests relies on your fritzbox setup and the number of devices (network/dect/etc) connected to it.

I close this issue for now, feel free to comment or open a new one if issues occur again
